lib/internal/Magento/Framework/Reflection/TypeProcessor.php

```diff
@@ -521,14 +522,78 @@ public function getParamType(ParameterReflection $param)
                 $param->getDeclaringFunction()->name
             ));
         }
-        if ($type == 'array') {
+        if ($type === 'array') {
             // try to determine class, if it's array of objects
             $paramDocBlock = $this->getParamDocBlockTag($param);
             $paramTypes = $paramDocBlock->getTypes();
             $paramType = array_shift($paramTypes);
+
+            $paramType = $this->resolveFullyQualifiedClassName($param->getDeclaringClass(), $paramType);
+
             return strpos($paramType, '[]') !== false ? $paramType : "{$paramType}[]";
         }
-        return $type;
+
+        return $this->resolveFullyQualifiedClassName($param->getDeclaringClass(), $type);
+    }
+
+    /**
+     * Resolve fully qualified type name in the class alias context
+     * @param ClassReflection $sourceClass
+     * @param string $typeName
+     * @return string
+     */
+    public function resolveFullyQualifiedClassName(ClassReflection $sourceClass, string $typeName): string
+    {
+        $typeName = trim($typeName);
+
+        // Not a class, but a basic type
+        if ((strtolower($typeName[0])) === $typeName[0]) {
+            return $typeName;
+        }
+
+        // Split array suffix
+        if (preg_match('/^(.+?)\[\]$/', $typeName, $matches)) {
+            $typeName = $matches[1];
+            $arraySuffix = '[]';
+        } else {
+            $arraySuffix = '';
+        }
+
+        // If class exists, then return it
+        try {
+            $classTypeReflection = new ClassReflection($typeName);
+            return $typeName . $arraySuffix;
+        } catch (\ReflectionException $e) {
+            unset($e);
+        }
+
+        // Resolve fully qualified name
+        $sourceFileName = $sourceClass->getDeclaringFile();
+        $source = $sourceFileName->getContents();
+        $parser = new TokenParser($source);
```

Comment:
Why go this route vs. using `$sourceFileName->getUses()` instead?

Reply:
You are right, I did not know about `getUses`. I am refactoring now. Many thanks.